### ISSUE TITLE
org-return-dwim doesn't work in bullet list

### ISSUE THREAD

**Issue by amosbird** [jkitchin/scimax#127]
```
* test
  - aaaa      <---- hit return here
*  <---- ends here
```

I'm using org 9.1 in spacemacs.

**Comment by amosbird** [comment]
also
```
* test
  - [ ] aaa
  - [ ]   <---- hit return here
  - [ ]  <--- generates a new item instead of clear the prefix
```

**Comment by et2010** [comment]
I also observed some weird behavior of `org-return-dwim`. So I switched to org-autolist: https://github.com/calvinwyoung/org-autolist and it does the job for me.

**Comment by jkitchin** [comment]
I think I have fixed the checkbox issue. I can't reproduce the list issue, it works as expected for me.

**Comment by jkitchin** [comment]
The list issue may be on the most modern org 9.1. I didn't see it in 9.0, but did on 9.1. I have pushed a fix related to issue #147 that probably fixes this.
